src/radio-group/radio-button.tsx

```diff
@@ -40,12 +36,21 @@ export default function RadioButton({
         <input
           {...nativeControlProps}
           type="radio"
+          ref={mergedRefs}
           name={name}
           value={value}
           checked={checked}
-          onChange={onChange && (() => fireNonCancelableEvent(onChange, { value }))}
+          // empty handler to suppress React controllability warning
+          onChange={() => {}}
```

Comment:
Events on the native element still work correctly, since they bubble up to the surrounding `.label-wrapper` element. We test this in our unit tests by using the `findInputByValue` test util.